### PR TITLE
[TT-XLA][PyTorch] Training Workload for GPT-OSS-120B-Galaxy

### PR DESCRIPTION
**Pull request by abogdanovicTT**
**Issue**

N/A

**Training Workload Setting**

  Model: GPT-OSS-120B
  Strategy: LoRA
  Framework: PyTorch
  Dataset: Alpaca
  Hardware: WH Galaxy

**What's Changed**

This PR introduces training of GPT-OSS-120B on WH Galaxy.

**Experiment Results**
<table>
  <tr>
    <td>Train Loss
    <td>Val Loss
  <tr>
  <tr>
    <td></td>
    <td></td>
  </tr>
</table>

**Checklist**

- [x]     Addition of appropriate .yaml file
- [x]     TT device testing
- [ ]     GPU testing

### CODE REVIEW

**Review by pglusacTT — APPROVED** [review]
Let's go!!!

File: blacksmith/experiments/torch/gpt_oss/lora/galaxy/gpt_oss_120b_alpaca.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# LoRA setup
+lora_r: 32
+lora_alpha: 64
+lora_target_modules: ["q_proj", "k_proj", "v_proj", "o_proj"]
+lora_task_type: "CAUSAL_LM"
+
+# Device settings
+mesh_shape: [4, 8]
+mesh_axis_names: ["batch", "model"]
+input_sharding_dim: null
```
Comment (agobeljicTT) [inline blacksmith/experiments/torch/gpt_oss/lora/galaxy/gpt_oss_120b_alpaca.yaml:27]: this is only tensor parallel?

**Review by agobeljicTT — APPROVED** [review]

File: blacksmith/experiments/torch/gpt_oss/README.md [hunk 2]
```diff
@@ -1,16 +1,17 @@
 # GPT-OSS with LoRA Experiment in TT-XLA
 
-This directory contains the code for the GPT-OSS 20B LoRA fine-tuning experiment in TT-XLA.
+This directory contains the code for the GPT-OSS 20B and 120B LoRA fine-tuning experiments in TT-XLA.
```
Comment (mmilosevicTT) [inline blacksmith/experiments/torch/gpt_oss/README.md]:
```suggestion
This directory contains the code for the GPT-OSS LoRA fine-tuning experiments in TT-XLA.
```

nit: We do not really need to repeat this all the time, it is present in the list below.

File: blacksmith/experiments/torch/gpt_oss/README.md [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 
 Original LoRA paper can be found [here](https://arxiv.org/pdf/2106.09685).
 
 ## Overview
 
-The GPT-OSS fine-tuning experiment applies the LoRA technique to adapt a pre-trained GPT-OSS 20B
-model on the SST-2 sentiment analysis dataset.
-The experiment is designed to run on the Huggingface framework.
+The GPT-OSS fine-tuning experiments apply the LoRA technique to adapt the pre-trained GPT-OSS 20B
+and 120B models on the SST-2 sentiment analysis and Alpaca instruction-tuning datasets.
```
Comment (mmilosevicTT) [inline blacksmith/experiments/torch/gpt_oss/README.md]:
```suggestion
The GPT-OSS fine-tuning experiments apply the LoRA technique to adapt pre-trained GPT-OSS
models on the SST-2 sentiment analysis and Alpaca instruction-tuning datasets.
```

nit: ditto for this

**Review by mmilosevicTT — APPROVED** [review]
Amazing job! 🎊  LGTM

Left few nit comments.
